src/backend/pipeline/stream.c

```diff
@@ -315,7 +434,17 @@ InsertIntoStreamPrepared(PreparedStreamInsertStmt *pstmt)
 	stream_stat_report(pstmt->relid, count, 1, size);
 
 	if (synchronous_stream_insert)
```

Comment:
Can we try to unify this waiting code into some refactored functions? I feel like this whole sync stream stuff is getting a little ugly with this change. Lemme think about how we can clean it up.